rust-runtime/smithy-async/src/rt/sleep.rs

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Provides an [`AsyncSleep`] trait that returns a future that sleeps for a given duration,
+//! and implementations of `AsyncSleep` for different async runtimes.
+
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Duration;
+
+/// Async trait with a `sleep` function.
+pub trait AsyncSleep: std::fmt::Debug + Send + Sync {
```

Comment:
My reasoning here was that someone would be able to tell which sleep implementation they were using by debug printing it, which would result in `TokioSleep` if it were the Tokio implementation provided in this same file. I suspect implementations of this won't have any data associated with them.

Reply:
I would be consistent w/ all other traits. It isn't unreasonable to require `Debug` when dealing mostly w/ trait objects. Most types implement `Debug`, so it shouldn't be much of a burden. However, if no other trait extends Debug, I probably wouldn't do it here either.

Reply:
This is the first and only trait in the crate so far, so it will be consistent. This crate is intended to be used to provide async runtime agnostic abstractions, so having `Debug` on the traits to be able to tell which concrete implementation is being used is probably valuable, especially since these traits will be used as trait objects.